com.microsoft.mrtk.input/Visualizers/RiggedHandVisualizer/RiggedHandMeshVisualizer.cs

```diff
@@ -0,0 +1,302 @@
+        }
+
+        [SerializeField]
+        [Range(0.8f, 1.2f)]
+        [Tooltip("The overall hand mesh will be scaled by this amount to fit the user's real hand size.")]
+        // Will be automatically calculated in the future.
+        private float handScale = 1.0f;
+
+        [SerializeField]
+        [Tooltip("The transform of the wrist joint.")]
+        private Transform wrist;
+
+        /// <summary>
+        /// Used to track whether the hand material has the appropriate property
+        /// </summary>
+        [SerializeField]
+        [HideInInspector]
+        private bool showMissingPropertyWarning = true;
+
+        /// <summary>
+        /// Used to track whether the hand renderer was provided
+        /// </summary>
+        [SerializeField]
+        [HideInInspector]
+        private bool showMissingRendererWarning = false;
+
+        [SerializeField]
+        [Tooltip("Renderer of the hand mesh")]
+        [ShowInfoIf(InfoType.Warning, "Rigged Mesh Renderer is missing", "showMissingRendererWarning")]
+        private SkinnedMeshRenderer handRenderer = null;
+
+        // The property block used to modify the pinch amount property on the material
+        private MaterialPropertyBlock propertyBlock = null;
+
+        /// <summary>
+        /// Property name for modifying the mesh's appearance based on pinch strength
+        /// </summary>
+        private const string pinchAmountMaterialProperty = "_PinchAmount";
+
+        // Caching local references 
+        private HandsAggregatorSubsystem handsSubsystem;
+
+        // Scratch list for checking for the presence of display subsystems.
+        private List<XRDisplaySubsystem> displaySubsystems = new List<XRDisplaySubsystem>();
+
+        private XRBaseController controller;
+
+        // The actual, physical, rigged joints that drive the skinned mesh.
+        // Otherwise referred to as "armature". Must be in OpenXR order.
+        private readonly Transform[] riggedVisualJointsArray = new Transform[(int)TrackedHandJoint.TotalJoints];
+
+        // Record the initial, intrinsic lengths of the tip bones of the skinned mesh.
+        // Used later to adjust the length of the skinned mesh's fingertips for improved accuracy.
+        private readonly float[] initialTipLengths = new float[(int)TrackedHandJoint.TotalJoints];
+
+        protected virtual void Awake()
+        {
+            propertyBlock = new MaterialPropertyBlock();
+
+            // Start the depth-first-traversal at the wrist index.
+            int index = (int)TrackedHandJoint.Wrist;
+
+            // This performs a depth-first-traversal of the armature. Ensure
+            // the provided armature's bones/joints are in OpenXR order.
+            foreach (Transform child in wrist.GetComponentsInChildren<Transform>())
+            {
+                // The "leaf joints" are excluded
+                if (child.name.Contains("end")) { continue; }
```

Comment:
Can we make these strings into constants rather than defined in-line

Reply:
Resolved